File: 70climbingStairs/climbingStairs.java
class Solution {
    public int climbStairs(int n) {
        return Stream.iterate(Arrays.asList(1, 1), result -> Arrays.asList(result.get(1), result.stream().mapToInt(Integer::intValue).sum())).mapToInt($ -> $.get(1)).limit(n).reduce((a,b) -> b).orElse(0);
    }
}

File: 70climbingStairs/climbingStairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        std::array result{1l, 1l};
        for (std::size_t _{}; _ != n; ++_) result = {result.back(), result.front() + result.back()};
        return result.front();
    }
};
